test/wait_for_blocks.go

```diff
@@ -28,6 +29,45 @@ func WaitForBlocks(ctx context.Context, delta int, chains ...ChainHeighter) erro
 	return eg.Wait()
 }
 
+// WaitForInSync blocks until all nodes have heights greater than or equal to the chain height.
+func WaitForInSync(ctx context.Context, chain ChainHeighter, nodes ...ChainHeighter) error {
+	if len(nodes) == 0 {
+		panic("missing nodes")
+	}
+InSyncLoop:
+	for {
+		if ctx.Err() != nil {
+			return fmt.Errorf("timed out waiting for node(s) to be in sync: %w", ctx.Err())
```

Comment:
nit:

Context.Err() may not be because of a timeout. It could've been cancelled for other reasons.

Reply:
deferred to the ctx.Err()